src/form-elements/header/edit-donation.ts

```diff
@@ -0,0 +1,402 @@
+        <ul class="amount-selector">
+          ${this.presetAmountsTemplate}
+          <li class="custom-amount">${this.customAmountTemplate}</li>
+        </ul>
+
+        <div class="errors">
+          ${this.error}
+        </div>
+
+        <div class="cover-fees-container">
+          <input type="checkbox" id="cover-fees" @input=${this.coverFeesChecked} />
+          <label for="cover-fees">
+            I'll generously add ${currency(this.donationInfo.fee, { symbol: '$' }).format()} to
+            cover the transaction fees so you can keep 100% of my donation.
+          </label>
+        </div>
+      </form-section>
+    `;
+  }
+
+  private get frequencyTemplate(): TemplateResult {
+    return html`
+      <li>
+        ${this.getRadioButton({
+          group: SelectionGroup.DonationType,
+          value: DonationType.OneTime,
+          displayText: 'One time',
+          checked: this.donationInfo.donationType === DonationType.OneTime,
+        })}
+      </li>
+
+      <li>
+        ${this.getRadioButton({
+          group: SelectionGroup.DonationType,
+          value: DonationType.Monthly,
+          displayText: 'Monthly',
+          checked: this.donationInfo.donationType === DonationType.Monthly,
+        })}
+      </li>
+    `;
+  }
+
+  private get presetAmountsTemplate(): TemplateResult {
+    return html`
+      ${this.amountOptions.map(amount => {
+        const checked = !this.customAmountButton?.checked && amount === this.donationInfo.amount;
+        return html`
+          <li>
+            ${this.getRadioButton({
+              group: SelectionGroup.Amount,
+              value: `${amount}`,
+              displayText: `$${amount}`,
+              checked: checked,
+            })}
+          </li>
+        `;
+      })}
+    `;
+  }
+
+  private getRadioButton(options: {
+    group: SelectionGroup;
+    value: string;
+    displayText: string;
+    checked: boolean;
+  }): TemplateResult {
+    const radioId = `${options.group}-${options.value}-option`;
+    return html`
+      <div class="selection-button">
+        <input
+          type="radio"
+          name=${options.group}
+          value=${options.value}
+          id=${radioId}
+          .checked=${options.checked}
+          @change=${this.radioSelected}
+        />
+        <label for=${radioId}>${options.displayText}</label>
+      </div>
+    `;
+  }
+
+  private get customAmountTemplate(): TemplateResult {
+    const isCustomAmount = !this.amountOptions.includes(this.donationInfo.amount);
+    const value = isCustomAmount ? this.donationInfo.amount : '';
+
+    return html`
+      <div class="selection-button">
+        <input
+          type="radio"
+          name=${SelectionGroup.Amount}
+          value="custom"
+          id="custom-amount-button"
+          @change=${this.customRadioSelected}
+        />
+
+        <label for="custom-amount-button">
+          <span class="custom-amount-text">Custom: $</span
+          ><input
```

Comment:
this is wild. an input in a label

Reply:
That's how I had always done it. You don't need the `for=` when you do it this way. Do we have a standard for that?